### ISSUE TITLE
Continue training with different number of points

### ISSUE THREAD

**Issue by ileniTudor** [yinguobing/cnn-facial-landmark#34]
Hello, thank for sharing your work.
After we train the model, with the facial 136 points, how can we continue the training with a different number of points, let say 60? 
The only layer I would change is :

`logits = tf.layers.dense(
        inputs=dense1,
        units=60, 
        activation=None,
        use_bias=True,
        name="logits")`

But the number of parameter of the network will be change. So, we cannot continue from the previous checkpoint. Who can we deal with it, programmatically?

If I try to restore the previous checkpoints (trained for the 136 facial points), to the new graph that outputs 60 points. I get the following error:
`InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Assign requires shapes of both tensors to match. lhs shape= [1024,60] rhs shape= [1024,136]`

How can I restore the checkpoint only for the previous layers, not for the Logits layer (that now has a different shape)?

**Comment by yinguobing** [comment]
This is similar to transfer learning which TensorFlow is capable of.
You can figure it out by reading the official ResNet model's implementation:
https://github.com/tensorflow/models/blob/4b4dbad1e983543f196cf7909710775059af7e92/official/resnet/resnet_run_loop.py#L574

Good luck!

**Comment by abdou31** [comment]
@ileniTudor how did you train the model ? what command you used for training ?

**Comment by ileniTudor** [comment]
> @ileniTudor how did you train the model ? what command you used for training ?

It took some time since I was playing with this code, but i remember doing something like this, when loading the checkpoints (In this way you can create your new shape logits layer, and not loading it from the checkpoints) :

    label_tensor = tf.convert_to_tensor(labels, dtype=tf.float32)
    my_exclude = ['IteratorGetNext', 'logits/BiasAdd', 'logits/bias', 'logits/kernel']
    variables_to_restore = tf.contrib.slim.get_variables_to_restore(exclude=my_exclude)
    scopes = {os.path.dirname(v.name) for v in variables_to_restore}
    tf.train.init_from_checkpoint('train-300vw/model.ckpt-29332',
                                  {s + '/': s + '/' for s in scopes})

If you want to train only the logit layer, and freez the other, try something like:

```
       optimizer = tf.train.AdamOptimizer(learning_rate=0.001)
       layer_training_vars = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES, "logits")
       train_op = optimizer.minimize(
                loss=loss,
                global_step=tf.train.get_global_step(),
                var_list=layer_training_vars)
```

**Comment by abdou31** [comment]
I trained the model with my own dataset using this tutorial:
+ https://github.com/faust690226/cnn-facial-landmark-tutorial
I get good results with Facial landmarks but using my own dataset, I get some bad results.
I want to know what are output and input nodes names that can be used for producing the .pb frozen model?

**Comment by ileniTudor** [comment]
@abdou31 I did'n do that yet. 
I ran the `cnn_model_fn ` with the debugger. The input name seems to be: input_to_float .The output is: logits/BiasAdd.

Try it by your own, for your graph. Or print the graph variables list.

**Comment by abdou31** [comment]
i set logits/BiasAdd to output nodes names but i don't that input nodes is recomanded or no frozing model?
